src/Core/Checkout/Payment/Controller/PaymentController.php

```diff
@@ -56,85 +64,170 @@ public function finalizeTransaction(Request $request): Response
     {
         $paymentToken = $request->get('_sw_payment_token');
 
-        if ($paymentToken === null) {
+        if (!\is_string($paymentToken)) {
             // @deprecated tag:v6.8.0 - remove this if block
             if (!Feature::isActive('v6.8.0.0')) {
                 throw RoutingException::missingRequestParameter('_sw_payment_token'); // @phpstan-ignore-line shopware.domainException
             }
             throw PaymentException::missingRequestParameter('_sw_payment_token');
         }
 
-        $token = $this->tokenFactory->parseToken($paymentToken);
+        try {
+            $token = null;
+            $oldToken = null;
+            if (Feature::isActive('v6.8.0.0')) {
+                $token = $this->paymentTokenGenerator->decode($paymentToken);
+            }
 
-        if (Feature::isActive('REPEATED_PAYMENT_FINALIZE') && $token->isConsumed()) {
-            $this->handleResponse($token);
-        }
+            $return = null;
+            Feature::callSilentIfInactive('v6.8.0.0', function () use ($paymentToken, &$token, &$oldToken, &$return): void {
+                if (!$this->tokenFactory) {
+                    // @phpstan-ignore-next-line
```

Comment:
would you mind changing it to this? 
```suggestion
                    // @phpstan-ignore specific.identifier.change.me! (comment why we need to ignore it)
```

also see https://phpstan.org/user-guide/ignoring-errors
ignore-next-line will ignore every PHPStan issue and not just the specific one.

Reply:
was unnecessary anyway ("vorauseilender Gehorsam"). I assumed there would be an issue because I'm throwing a non-domain-exception, but PHPstan doesn't complain